### PR TITLE
New package openPMD_api

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by giordano** [comment]
For musl you either need to ship [this patch](https://github.com/JuliaPackaging/Yggdrasil/blob/f84926f3156eb65d245477cc4ed92d0f516199cb/Z/ZeroMQ/bundled/patches/mm_malloc.patch) or require GCC 6 or 7 (I never remember when this was fixed)

**Comment by eschnett** [comment]
Thanks. I'm switching to GCC 6 anyway because of a `std::to_string` bug on FreeBSD in GCC <6.

Isn't `mm_malloc.h` a system header file? Applying a patch to it seems adventurous. Long live Docker and isolated build environments!

**Comment by giordano** [comment]
> Isn't mm_malloc.h a system header file? Applying a patch to it seems adventurous.

Yeah, in fact one should do something like this:
https://github.com/JuliaPackaging/Yggdrasil/blob/f84926f3156eb65d245477cc4ed92d0f516199cb/Z/ZeroMQ/build_tarballs.jl#L19-L25
Otherwise use a version of GCC where they finally implemented the function correctly for Musl

File: O/openPMD_api/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+mpiopts=
+if [[ "$target" == x86_64-w64-mingw32 ]]; then
+    mpiopts="-DMPI_HOME=$prefix -DMPI_GUESS_LIBRARY_NAME=MSMPI -DMPI_C_LIBRARIES=msmpi64 -DMPI_CXX_LIBRARIES=msmpi64"
+elif [[ "$target" == *-mingw* ]]; then
+    mpiopts="-DMPI_HOME=$prefix -DMPI_GUESS_LIBRARY_NAME=MSMPI"
+fi
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake -DCMAKE_BUILD_TYPE=Release ${mpiopts} ..
+make -j${nproc}
+make -j${nproc} install
+"""
```
Comment (giordano) [inline O/openPMD_api/build_tarballs.jl:38]: The license files aren't installed automatically:
```suggestion
install_license ../COPYING*
"""
```

**Pull request merged**